fix(toc): fixed jumplinks in TOC to resolve a11y nesting issue

evwilkin opened the pull request:
closes #2969 

This PR:
- Mirrors the structure of the [nested jump links example](https://www.patternfly.org/v4/components/jump-links#expandable-vertical-with-subsection) from our docs to fix a11y error of a `<ul>` nested directly inside another `<ul>`.

Fix:
Previously we were either rendering `<JumpLinksList>` directly inside the `<JumpLinks>` component:
```
<JumpLinks>
  <JumpLinksItem><JumpLinksItem>
  <JumpLinksList><JumpLinksList/>
</JumpLinks>
```

We should always render a `<JumpLinksItem>` and place any needed `<JumpLinksList>`s within them where needed.
```
<JumpLinks>
  <JumpLinksItem></JumpLinksItem>
  <JumpLinksItem>
    <JumpLinksList><JumpLinksList/>
  </JumpLinksItem>
</JumpLinks>
```

patternfly-build commented:
PF4 preview: https://patternfly-org-pr-2971-v4.surge.sh/v4

thatblindgeye commented on packages/theme-patternfly-org/components/tableOfContents/tableOfContents.js:
```diff
@@ -10,25 +10,57 @@ export const TableOfContents = ({ items }) => {
-      ? (
-        <JumpLinksList key={index} className="ws-toc-sublist">
-          {item.map(renderItem)}
+  const renderSublist = (item, nextItem, idx) => {
```
For this, does `idx` need to be utilized now that the key attribute is no longer using the index, or could the index be omitted from the params here?

evwilkin replied:
You're correct, this was leftover from previous refactoring - removed this unused param 👍

thatblindgeye commented on packages/theme-patternfly-org/components/tableOfContents/tableOfContents.js:
```diff
@@ -10,25 +10,57 @@ export const TableOfContents = ({ items }) => {
+
+  const renderJumpLinksItems = () => {
+    items.forEach((item, index) => {
+      let cur = item;
```
Should `cur` be utilized anywhere within this function, or could this line be removed?

evwilkin replied:
Same as above - removed this unused var 👍

thatblindgeye reviewed (commented):
Looks good in the preview! I just had a couple of questions below if you wouldn't mind clarifying as I wasn't 100% sure.

evwilkin commented:
@thatblindgeye thanks for the feedback, I've removed the unused var/param you called out and made small name changes to other vars to be more descriptive in [my second commit](https://github.com/patternfly/patternfly-org/pull/2971/commits/e5f26aaf0f4c21326075bda2c89877a4e081496a):
- renamed `next` to `nextItem`
- renamed `nextItem` where it's passed as an array to `nextItemArr`
